### PR TITLE
Feature/remove poetry from pytorch example

### PR DESCRIPTION
**Pull request by chongshenng**
### Description

`poetry` and `pip` does not manage the environments consistently. This PR removes `poetry` from the `quickstart-pytorch/` example.

### Related issues/PRs

Supercedes #3222 

### Explanation

Use `pyproject.toml` to manage dependencies and install with the `hatchling` backend.

### Checklist

- [X] Implement proposed change
- [X] Update the changelog entry below
- [X] Make CI checks pass
- [X] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### Changelog entry

<examples>

### CODE REVIEW

File: examples/quickstart-pytorch/pyproject.toml [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 
-[tool.poetry.dependencies]
-python = ">=3.8,<3.11"
-flwr = ">=1.8.0,<2.0"
-flwr-datasets = { extras = ["vision"], version = ">=0.0.2,<1.0.0" }
-torch = "2.1.1"
-torchvision = "0.16.1"
-tqdm = "4.65.0"
+[tool.hatch.build.targets.wheel]
+packages = ["."]
```
Comment (charlesbvll) [inline examples/quickstart-pytorch/pyproject.toml:21]: You're missing a new line in this file

File: examples/advanced-pytorch/server.py [hunk 2]
```diff
@@ -10,8 +10,6 @@
 
 import warnings
 
-from flwr_datasets import FederatedDataset
-
```
Comment (charlesbvll) [inline examples/advanced-pytorch/server.py:14]: This should be moved to another PR ideally

File: examples/quickstart-pytorch/pyproject.toml (hunk 1 above)
Comment (chongshenng) [reply examples/quickstart-pytorch/pyproject.toml:21]: Are you suggesting a new line at the end of this file? Just curious, what's the reasoning for it (consistency?)?

File: examples/advanced-pytorch/server.py (hunk 2 above)
Comment (chongshenng) [reply examples/advanced-pytorch/server.py:14]: Sure, will remove it :)

File: examples/quickstart-pytorch/pyproject.toml (hunk 1 above)
Comment (charlesbvll) [reply examples/quickstart-pytorch/pyproject.toml]: I think this is just good practice because otherwise it might mess up the formatting in some scenarios: https://stackoverflow.com/a/5813359. Btw, this is the reason for the red circle with a minus inside it, displayed by GitHub

Comment (charlesbvll) [reply examples/quickstart-pytorch/pyproject.toml]: @chongshenng Here is a simple example I made to showcase the difference

image

**Review by charlesbvll — APPROVED** [review]
LGTM!

**Review by tanertopal — APPROVED** [review]
LGTM
